feat(pie-docs): DSW-1484 icon button overview page (#993)
justeattakeaway/pie: apps/pie-docs/src/components/icon-button/overview/overview.md

---
eleventyNavigation:
    key: Overview
    parent: Icon Button
    order: 1
eleventyComputed:
    sizes: "{% include './sizes.json' %}"
shouldShowContents: true
---

## Overview

The purpose of icon buttons is to offer a visually engaging and efficient way for users to interact with various functions or actions within an interface. They allow for clear and concise communication of functionality, particularly in situations where text may be less practical or necessary.

Icon buttons feature simple and recognizable icons that represent a specific action or functionality. They are often used in toolbars, navigation menus, or interactive elements where space is limited or a visual representation is more intuitive than text.

{% contentPageImage {
    src:"../../../assets/img/components/icon-button/overview.svg",
    alt: "Two icon buttons that are displayed together in a row."
} %}

---

## Dos and Don’ts

{% usage {
    do: {
        type: usageTypes.text,
        items: [

>            "Use it when an icon button on it’s own can clearly indicate the action."

Reviewer: ```suggestion
           "Use it when an icon button on its own can clearly indicate the action."
```

        ]
    },
    dont: { 
        type: usageTypes.text,
        items: [

>             "Don’t use if an icon on it’s own doesn’t clearly indicate the action, in the case use a standard [Button](/components/button/overview)."

Reviewer: ```suggestion
            "Don’t use if an icon on its own doesn’t clearly indicate the action, in the case use a standard [Button](/components/button/overview)."
```

    } %}
  {% endcontentItem %}
  {% contentItem %}
    <h3>Loading</h3>
    {% contentPageImage {
      src: "../../../assets/img/components/icon-button/interactive-state-loading.svg",
      width: 48,
      alt: "An icon button that is loading."
    } %}
  {% endcontentItem %}
{% endcontentLayout %}

---

## Example

Outlines the atomic level interactive elements for the component.

{% contentLayout %}
  {% contentItem %}
    {% contentPageImage {
      src: "../../../assets/img/components/icon-button/example-hamburger-menu.svg",
      width: 48,
      alt: "A hamburger menu icon button."
    } %}
  {% endcontentItem %}
  {% contentItem %}
    {% contentPageImage {
      src: "../../../assets/img/components/icon-button/example-close.svg",
      width: 48,
      alt: "A close icon button."
    } %}
  {% endcontentItem %}
  {% contentItem %}
    {% contentPageImage {
      src: "../../../assets/img/components/icon-button/example-icon-button-pair.svg",
      width: 104,
      alt: "Info and heart icon buttons."
    } %}
  {% endcontentItem %}
{% endcontentLayout %}

--- 

## Resources

{% resourceTable {
    rows: [
        {
            resource: resourceTypes.COMPONENT,

>             link: "https://www.figma.com/file/R2rBfzJP0hG0MZorq6FLZ1/%5BCore%5D-Components-%E2%9A%AA%EF%B8%8F-%5BPIE-2.0%5D?type=design&node-id=981-5350&mode=design",

Reviewer: This link takes me to a Form label?

Author: updated

            status: statusTypes.AVAILABLE
        },
        {
            resource: resourceTypes.DOCUMENTATION,
            link: "https://www.figma.com/file/j1YKygEyhqZ6zKVxcHapn5/%5BCore%5D-Component-Documentation-%E2%9A%AA%EF%B8%8F-%5BPIE-2.0%5D?type=design&node-id=742-317&mode=design",
            status: statusTypes.AVAILABLE
        },
        {
            resource: resourceTypes.WEB_COMPONENTS,
            link: "https://webc.pie.design/?path=/story/icon-button--primary",

>             status: statusTypes.ALPHA

Reviewer: In the designs it looks like this page only has 5 links (Figma x2, Webc, Fozzie and Snacks) and they are all `AVAILABLE`

Author: yes but the web component status is still in ALPHA so we can't move to available.
